tests/sentry/utils/test_event_frames.py

```diff
@@ -376,6 +378,74 @@ def test_not_munged(self):
         assert not munged_frames
 
 
+class FlutterFilenameMungingTestCase(TestCase):
+    def test_not_munged(self):
+        frames = [
+            {
+                "package": "my_package",
+                "filename": "service.dart",
+                "abs_path": "package:my_package/a/b/service.dart",
+                "in_app": True,
+            }
+        ]
+        munged_frames = munged_filename_and_frames("other", frames, "doesnt_matter", "sentry.sdk")
+        assert not munged_frames
+
+    def test_flutter_munger_supported(self):
+        frames = [
+            {
+                "function": "tryCatchModule",
+                "package": "sentry_flutter_example",
+                "filename": "test.dart",
+                "abs_path": "package:sentry_flutter_example/a/b/test.dart",
+                "lineno": 8,
+                "colno": 5,
+                "in_app": True,
+            }
+        ]
+        munged_frames = munged_filename_and_frames(
+            "other", frames, "munged_filename", "sentry.dart.flutter"
+        )
+        munged_first_frame: Mapping[str, Any] = munged_frames[1][0]
+        assert munged_first_frame.items() > frames[0].items()
+        assert munged_first_frame["munged_filename"] == "a/b/test.dart"
+
+    def test_dart_prefix_not_munged(self):
+        assert not flutter_frame_munger(
+            "munged_filename",
+            {
+                "abs_path": "dart:ui/a/b/test.dart",
+            },
+        )
+
+    def test_abs_path_not_present_not_munged(self):
+        assert not flutter_frame_munger(
+            "munged_filename",
+            {
+                "function": "tryCatchModule",
+                "package": "sentry_flutter_example",
+                "filename": "test.dart",
+            },
+        )
+
+    def test_different_package_not_munged(self):
+        assert not flutter_frame_munger(
+            "munged_filename",
+            {
+                "package": "sentry_flutter_example",
+                "abs_path": "package:different_package/a/b/test.dart",
```

Comment:
Is this case just so that we don't munge file paths from 3rd party packages and accidentally match to them?

Reply:
Tbh, this was more of a potential edge-case I thought up. I don't know if this could actually happen in the wild. 

From the examples I've seen, the `package` value should match whatever is in `abs_path=package:<package>/a/b/test.dart`. 

I specifically test for this to be pessimistic on path frame munging to avoid undefined behavior and introducing weird munged filepaths.